source/utilities.cc

```diff
@@ -362,15 +362,15 @@ namespace aspect
     }
 
 
-
     std::vector<double>
     parse_map_to_double_array (const std::string &input_string,
                                const std::vector<std::string> &list_of_keys,
                                const bool expects_background_field,
                                const std::string &property_name,
                                const bool allow_multiple_values_per_key,
                                const std::unique_ptr<std::vector<unsigned int>> &n_values_per_key,
-                               const bool allow_missing_keys)
+                               const bool allow_missing_keys,
```

Comment:
If you keep `allow_missing_keys` last, it saves a function argument for all the calls to this function that you changed.

Reply:
You are right about those entries in the flow laws (e.g. material_model/rheology/diffusion_creep.cc), where the values for allow_missing_keys don't need to be specified. But in the material_model/utilities.cc, there are places where the "allow_missing_keys" need to be "True", and an additional entry of "use_values_instead_of_differences" needs to be added there. Moreover, if I modify this, I need to consider specifically about the situation where "allow_missing_keys" need to be "True", and then add an entry accordingly. I would go with the current implementation, as this follows the order by which these entries are entered. It might seem messy in some places, but we are also staying on the safe side.

Reply:
1. The name of the parameter
I have changed it to the suggested name.
2. The parse function can keep allow_missing_keys as last argument, so that you don't need to give it as argument each time you call the function.
I have commented in the conversation.
3. Do all the phase parameters need the option to be passed as differences? If I remember correctly, we discussed the option for the density and there I think it makes sense. But for friction angles, for example, do you see a use case?
It would also be helpful with the friction angles. Let's say, we want to assign a weak zone that goes to 100 km depth. We can achieve this by making a shallower phase and a deeper phase. Then we can assign a smaller value of the friction angle to the shallower phase.
4. Why did you choose for some parameters to add logarithmically and for some not? This needs comments. Also, should it be user-specified?
That's a very good point. And what I actually did is to use quotient in this PR where the values are entered by logarithmic in the phase properties. This mainly accounts for the prefactors in the flow laws. The main reason is these values could differ by magnitude between phases. For example, if 1e-10, and 1e-15 are two values for two adjacent phases, it's much easier to enter by 1e-5 rather than (1e-15 - 1e-10).

Reply:
@anne-glerum, thanks for your review. I have addressed all the comments. Please feel free to tell me what I need to do for the next step.